cmd/skopeo/sync.go

```diff
@@ -50,9 +50,10 @@ type syncOptions struct {
 
 // repoDescriptor contains information of a single repository used as a sync source.
 type repoDescriptor struct {
-	DirBasePath string                 // base path when source is 'dir'
-	ImageRefs   []types.ImageReference // List of tagged image found for the repository
-	Context     *types.SystemContext   // SystemContext for the sync command
+	DirBasePath    string                 // base path when source is 'dir'
+	DestinationRef types.ImageReference   // Destination
```

Comment:
`types.ImageReference` inherently points at a _single_ image. It should not be abused as a repo/directory value.

One way to do this might be to move the `destinationReference` calls into `repoDescriptor` construction somehow, and to have `repoDescriptor` already contain fully-resolved (source reference, destination reference) pairs.

Reply:
It would be nice to try and render everything down to pairs of source and destination references regardless of how sync was called i.e. using specific transports from the command line or using one of the yaml configs. I think this is possible however it could be a rather large change